packages/block-editor/src/components/block-card/index.js

```diff
@@ -27,38 +31,52 @@ function BlockCard( { title, icon, description, blockType, className } ) {
 		( { title, icon, description } = blockType );
 	}
 
-	const { parentNavBlockClientId } = useSelect( ( select ) => {
-		const { getSelectedBlockClientId, getBlockParentsByBlockName } =
-			select( blockEditorStore );
+	const parentClientId = useSelect( ( select ) => {
+		const {
+			getSelectedBlockClientId,
+			getBlockParentsByBlockName,
+			__unstableGetContentLockingParent,
+		} = select( blockEditorStore );
 
 		const _selectedBlockClientId = getSelectedBlockClientId();
+		let _parentClientId = getBlockParentsByBlockName(
+			_selectedBlockClientId,
+			'core/navigation',
+			true
+		)[ 0 ];
 
-		return {
-			parentNavBlockClientId: getBlockParentsByBlockName(
-				_selectedBlockClientId,
-				'core/navigation',
-				true
-			)[ 0 ],
-		};
-	}, [] );
+		if ( ! _parentClientId ) {
+			_parentClientId = __unstableGetContentLockingParent(
+				_selectedBlockClientId
+			);
+		}
 
+		return _parentClientId;
+	}, [] );
+	const parentDisplayInfo = useBlockDisplayInformation( parentClientId );
+	const contentOnlyFills = useSlotFills( 'InspectorControlsContentOnly' );
 	const { selectBlock } = useDispatch( blockEditorStore );
 
 	return (
 		<div className={ classnames( 'block-editor-block-card', className ) }>
-			{ parentNavBlockClientId && ( // This is only used by the Navigation block for now. It's not ideal having Navigation block specific code here.
-				<Button
-					onClick={ () => selectBlock( parentNavBlockClientId ) }
-					label={ __( 'Go to parent Navigation block' ) }
-					style={
-						// TODO: This style override is also used in ToolsPanelHeader.
-						// It should be supported out-of-the-box by Button.
-						{ minWidth: 24, padding: 0 }
-					}
-					icon={ isRTL() ? chevronRight : chevronLeft }
-					size="small"
-				/>
-			) }
+			{ !! contentOnlyFills?.length &&
+				parentClientId && ( // This is only used by the Navigation block for now. It's not ideal having Navigation block specific code here.
+					<Button
+						onClick={ () => selectBlock( parentClientId ) }
```

Comment:
I wonder if the focus should return to the block itself or the parent `<NavigationItem>` instead.